src/main/c/waf_jni.c

```diff
@@ -1111,6 +867,131 @@ static bool _cache_single_class_weak(JNIEnv *env,
     return true;
 }
 
+
+static ddwaf_builder _get_builder_checked(JNIEnv *env, jclass clazz,
+                                                   jobject builder_obj)
+{
+    UNUSED(clazz);
+
+    ddwaf_builder builder = (ddwaf_builder)(intptr_t) JNI(GetLongField, builder_obj,
+                                            _builder_ptr);
+    if (JNI(ExceptionCheck)) {
+        return NULL;
+    }
+
+    if (!builder) {
+        JNI(ThrowNew, jcls_rte, "The Builder has already been cleared");
+    }
+
+    return builder;
+}
+
+JNIEXPORT void JNICALL Java_com_datadog_ddwaf_WafBuilder_removeConfigNative(JNIEnv *env, jclass clazz,
+                                                                        jobject builder, jstring path) {
+
+    UNUSED(clazz);
+    UNUSED(env);
+    if (builder && path) {
+        ddwaf_builder_remove_config(_get_builder_checked(env, clazz, builder), JNI(GetStringUTFChars, path, NULL), JNI(GetStringLength, path));
+    }
+}
+
+JNIEXPORT jboolean JNICALL Java_com_datadog_ddwaf_WafBuilder_addOrUpdateConfigNative(JNIEnv *env, jclass clazz,
+jobject builder, jstring path, jobject configuration, jobject diagnostics) {
+    ddwaf_object ddwaf_diagnostics;
+    ddwaf_object_invalid(&ddwaf_diagnostics);
+        struct _limits limits = {
+                .max_depth = 20,
+                .max_elements = 1000000,
+                .max_string_size = 1000000,
+        };
+    ddwaf_object ddwaf_configuration =_convert_checked(env, configuration, &limits, 0);
+    if (JNI(ExceptionCheck)) {
+        return JNI_FALSE;
+    }
+    uint32_t path_length = JNI(GetStringLength, path);
+    if (JNI(ExceptionCheck)) {
+        return JNI_FALSE;
+    }
+    ddwaf_builder ddwaf_builder = _get_builder_checked(env, clazz, builder);
```

Comment:
Eception check here.

Reply:
Also it's awkward that _get_builder_checked call is in the middle of the GetStringLength and GetStringUTFChars, as both are related to each other. Let's put the builder call either before or after both.